pkg/server/service.go

```diff
@@ -0,0 +1,45 @@
+package server
+
+import (
+	"context"
+
+	"github.com/grafana/dskit/services"
+
+	"github.com/grafana/grafana/pkg/api"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type service struct {
+	*services.BasicService
+	cla     setting.CommandLineArgs
+	opts    Options
+	apiOpts api.ServerOptions
+	server  *Server
+}
+
+func NewService(cla setting.CommandLineArgs, opts Options, apiOpts api.ServerOptions) (*service, error) {
```

Comment:
@mildwonkey this nested server feels a little weird to me

Reply:
I can make a pass at unwinding that (somehow 🤷 )